src/changelog/comment_changes.rs

```diff
@@ -280,7 +287,29 @@ impl Logic {
             },
             |r| {
                 self.repository = Some(r);
-                Ok(())
+
+                if let Some(tag) = &self.cli.tag {
+                    if let Some(repository) = &self.repository {
+                        if let Ok(target) =
+                            repository.resolve_reference_from_short_name(tag)
+                        {
+                            if target.is_tag() {
+                                self.stop_at_tag_oid = target.target();
+                                Ok(())
```

Comment:
If a tag name is given, it needs to be resolved such that the associated commit number can be used as additional stop condition.